core/commands/dag/stat.go

```diff
@@ -38,11 +39,22 @@ func dagStat(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment)
 	}
 
 	dagstats := &DagStat{}
+	isMHAlreadySeen := make(map[string]struct{})
 	err = traverse.Traverse(obj, traverse.Options{
 		DAG:   nodeGetter,
 		Order: traverse.DFSPre,
 		Func: func(current traverse.State) error {
-			dagstats.Size += uint64(len(current.Node.RawData()))
+			c := current.Node.Cid()
+			// Only count once blocks that show up multiple times with different codecs
```

Comment:
This would be incorrect as a default. There are a lot of codepaths out there ( including some of mine ) that use `ipfs dag stat` to approximate how big `ipfs dag export` would be. If you do the above then mixed cid version dags will not estimate correctly ( first hand experience - they do exist ).

Please find a way to reconcile `dag stat` and `dag export`

Reply:
> This would be incorrect as a default. There are a lot of codepaths out there ( including some of mine ) that use `ipfs dag stat` to approximate how big `ipfs dag export` would be. If you do the above then mixed cid version dags will not estimate correctly ( first hand experience - they do exist ).

My thinking is that `ipfs dag stat` should approximate the size taken by `ipfs pin add`.

Your use case sounds usefull tho, I think an optin flag that would enable deduping based on multihash would be nice.

Also why aren't `.car` files storing blocks keyed by multihash so they could also dedup things like that ? But that an other conversation.